shell/platform/fuchsia/flutter/accessibility_bridge.cc

```diff
@@ -82,9 +82,31 @@ AccessibilityBridge::GetNodeAttributes(const flutter::SemanticsNode& node,
 fuchsia::accessibility::semantics::States AccessibilityBridge::GetNodeStates(
     const flutter::SemanticsNode& node) const {
   fuchsia::accessibility::semantics::States states;
-  if (node.HasFlag(flutter::SemanticsFlags::kHasCheckedState)) {
-    states.set_checked(node.HasFlag(flutter::SemanticsFlags::kIsChecked));
+
+  // Set checked state.
+  if (!node.HasFlag(flutter::SemanticsFlags::kHasCheckedState)) {
+    states.set_checked_state(
+        fuchsia::accessibility::semantics::CheckedState::NONE);
+  } else {
+    states.set_checked_state(
+        node.HasFlag(flutter::SemanticsFlags::kIsChecked)
+            ? fuchsia::accessibility::semantics::CheckedState::CHECKED
+            : fuchsia::accessibility::semantics::CheckedState::UNCHECKED);
   }
+
+  // Set selected state.
+  states.set_selected(node.HasFlag(flutter::SemanticsFlags::kIsSelected));
+
+  // Set hidden state.
+  states.set_hidden(node.HasFlag(flutter::SemanticsFlags::kIsHidden));
+
+  // Set value.
+  std::string value = node.value;
+  if (value.size() > fuchsia::accessibility::semantics::MAX_VALUE_SIZE) {
+    value.resize(fuchsia::accessibility::semantics::MAX_VALUE_SIZE);
+  }
+  states.set_value(value);
```

Comment:
1. We should add a TODO for this, tied to fxb/17602 (aka fxb/mi4-2531).  We're potentially losing data here - although the value string is typically not very long.
2. We should avoid copying the string. We also need to now track how much size we've added on. Something like this will work:

```cpp
GetNodeStates(const flutter::SemanticsNode* node, size_t* additional_size) const {
  additionalSize += sizeof(fuchsia::accessibility::semantics::States);
...

  if (node.value.size() > fuchsia::accessibility::semantics::MAX_VALUE_SIZE) {
    states.set_value(node.value.substr(0, fuchsia::accessibility::semantics::MAX_VALUE_SIZE);
    *additional_size += fuchsia::accessibility::semantics::MAX_VALUE_SIZE;
  } else {
    states.set_value(node.value);
    *additional_size += node.value.size();
  }
  return states;
}
```

And we need to update the caller down below to pass in `&this_node_size` to this method now.

Reply:
My mistake -- I started to write the tests differently on my first pass, and forgot to change the flags value. Good catch.